kafka-scheduler/src/main/java/com/mesosphere/dcos/kafka/scheduler/KafkaScheduler.java

```diff
@@ -310,25 +304,44 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
                 status.getTaskId().getValue(),
                 status.getState().toString(),
                 status.getMessage()));
-
-        // Store status, then pass status to StageManager => Plan => Blocks
+        // Store status, then pass status to StageManager => Plan => Steps
         try {
             frameworkState.updateStatus(status);
         } catch (Exception e) {
             log.warn("Failed to update TaskStatus received from Mesos. "
                     + "This may be expected if Mesos sent stale status information: " + status, e);
         }
-
         planManager.update(status);
+        repairPlanManager.update(status);
+        reconciler.update(status);
 
-        if (hasOperations()) {
+        if (needsRecovery(status) || hasOperations()) {
             reviveOffers(driver);
         }
     }
 
+    private  boolean needsRecovery(TaskStatus taskStatus) {
```

Comment:
Should just use `TaskUtils.needsRecovery()`: https://github.com/mesosphere/dcos-commons/blob/0.8.1.1/sdk/scheduler/src/main/java/org/apache/mesos/offer/TaskUtils.java#L96